2-mandatory/3-shopping-list.js

```diff
@@ -19,7 +19,13 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
+  let name = recipe.name;
+  let recipeIngredients = recipe.ingredients;
+  let items = recipeIngredients.filter(
+    (e) =>
+      !pantry.cupboardContents.includes(e) && !pantry.fridgeContents.includes(e)
+  );
+  return (shoppingList = { name, items });
```

Comment:
It looks like you are using the shorthand for creating a new object. If you haven't already - you can read more about it here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#new_notations_in_ecmascript_2015
One question: do you need the shoppingList variable here? Maybe you can just return the object: `return { name, items };`?